inc/init.php

```diff
@@ -495,7 +495,11 @@ function getBaseURL($abs = null)
 
     //split hostheader into host and port
     if (isset($_SERVER['HTTP_HOST'])) {
-        $parsed_host = parse_url('http://' . $_SERVER['HTTP_HOST']);
+    	$cur_host = $_SERVER['HTTP_HOST'];
+    	if(isset($_SERVER['HTTP_X_FORWARDED_HOST'])) {
+    		$cur_host = $_SERVER['HTTP_X_FORWARDED_HOST'];
+    	}
```

Comment:
More compact doing effectively the same thing without the unnecessary assign if $_SERVER['HTTP_X_FORWARDED_HOST'] is set:
`$cur_host = $_SERVER['HTTP_X_FORWARDED_HOST'] ?? $_SERVER['HTTP_HOST'];`